### ISSUE TITLE
[BUG] cannot access newly generated dashboard

### ISSUE THREAD

**Issue by vitaly107** [avo-hq/avo#416]
I generated new dashborad as per the docs:

`rails generate avo:tool dashboard`

however when I try to access it in the browser I see an error:

```
NoMethodError at /uk/bj-admin/dashboard

undefined method `join' for nil:NilClass
Hint: Something is `nil` when it probably shouldn't be.
```

and ot point to line 65 here:

`
merge_script_names
actionpack (6.0.3.6) lib/action_dispatch/routing/routes_proxy.rb
60 61 62 63 64 65 66 67 68 69

        resolved_parts = new_script_name.count("/")

        previous_parts = previous_script_name.count("/")

        context_parts = previous_parts - resolved_parts + 1

        (previous_script_name.split("/").slice(0, context_parts).join("/")) + new_script_name

      end

    end

  end

end

`

**Comment by vitaly107** [comment]
` **(previous_script_name.split("/").slice(0, context_parts).join("/")) + new_script_name**`

**Comment by adrianthedev** [comment]
I really can't tell what is wrong from the information you gave me. 
Please show me what was added in your `routes.rb` file after you generated the tool.

**Comment by vitaly107** [comment]
as you see the error is not standard routing error and the route I am trying to access seems to be correct

the url I am visiting:

`http://localhost:3016/uk/bj-admin/dashboard`

routes.rb:

```
Rails.application.routes.draw do
  mount Avo::Engine => Avo.configuration.root_path
  root to: redirect('/uk'), as: :redirected_root

  scope :uk do

    scope 'bj-admin' do
      get 'dashboard', to: 'avo/tools#dashboard'
    end

  end

end
```

**Comment by vitaly107** [comment]
and in avo.rb I have

`config.root_path = '/uk/bj-admin'`

**Comment by adrianthedev** [comment]
Ok. I see the error. It's because the `root_path` is composed by two segments `uk` and `bj-admin`.

It's a bug with Avo and we'll fix it.

**Comment by vitaly107** [comment]
Yes please and then you may also wish to add something to the docs to provide guidance to people looking to access avo admin using some scope that is domain.com/scope/avo rather than domain.com/avo. Many thanks in advance.

**Comment by adrianthedev** [comment]
Sure. That makes sense!

**Comment by adrianthedev** [comment]
Hello @vitaly107,

I'm getting back to you with this bug. We're still working on a fix but we made some headway. It seems this is a bug with Rails with engines mounted under nested scopes.
As a quick fix you can do the following: In your routes file move the engine mount below the newly created tool and leave the `root_path` config to be `/uk/bj-admin`.

Ex:

```ruby
Rails.application.routes.draw do
  # other routes
  authenticate :user, ->(user) { user.is_admin? } do
    scope :uk do
      scope 'bj-admin' do
        get "dashboard", to: "avo/tools#dashboard"
      end
    end

    mount Avo::Engine, at: Avo.configuration.root_path
  end
end
```

```ruby
Avo.configure do |config|
  config.root_path = "/uk/bj-admin"
end
```

**This is just a quick fix.** 
We need to push some changes in order to properly fix this issue. We'll come back with instructions for nested engine routes.

**Comment by adrianthedev** [comment]
Released in [1.7.3](https://rubygems.org/gems/avo/versions/1.7.3)
